upstairs/src/lib.rs

```diff
@@ -4517,20 +4494,45 @@ impl Downstairs {
                         cdt::up__to__ds__read__done!(|| job.guest_id);
                     } else {
                         /*
-                         * If another job has finished already, we can
-                         * compare our read hash to
-                         * that and verify they are the same.
+                         * If another job has finished already, we can compare
+                         * our list of read hashes to that and verify they are
+                         * the same.
                          */
                         debug!(
                             self.log,
                             "[{}] Read already AckReady {}",
                             client_id,
                             job.ds_id
                         );
-                        if job.read_response_hashes != read_response_hashes {
+
+                        let mismatch = std::iter::zip(
```

Comment:
This code duplicates stuff in the `ack_status == AckStatus::Acked` condition above.  Perhaps it should be a function on `job`, e.g. `DownstairsIO::check_read_response_hashes(&self, read_data: &[ReadResponse]))`?

Reply:
I like it, also added in 6eb6d22

Reply:
I was thinking about also moving the rest of the stuff (including the logging and panic) into that function, but don't feel strongly about it.